### PR TITLE
AMRNAV-4736 support launching as composable node

### PR DESCRIPTION
**Pull request by jplapp**
Support launching as ComposableNode for IPC, and also zero-copy for the pointclouds.

Did not test whether this breaks normal launching, that should still be done before upstreaming

Inspired from realsense repo: 
- https://github.com/IntelRealSense/realsense-ros/pull/2279
- https://github.com/IntelRealSense/realsense-ros/pull/2599

### CODE REVIEW

File: src/k4a_ros_device.cpp [hunk 1]
```diff
@@ -281,6 +283,22 @@ K4AROSDevice::K4AROSDevice()
...
+    RCLCPP_ERROR_STREAM(this->get_logger(),"Failed to start cameras");
+  }
+
+  result = this->startImu();
+  if (result != K4A_RESULT_SUCCEEDED)
+  {
+    RCLCPP_ERROR_STREAM(this->get_logger(),"Failed to start IMU");
+  }
+
+  RCLCPP_INFO(this->get_logger(),"K4A Started");
```
Comment (jplapp) [inline src/k4a_ros_device.cpp:300]: this logic was in the `k4a_ros_bridge_node` , which manually created the executor in some main function. This is not supported in composable nodes. So I moved everything here and stopped building that node. In general, the new way should be fine also for launching in non-composable way.

File: CMakeLists.txt [hunk 2]
```diff
@@ -40,13 +41,10 @@ find_package(diagnostic_msgs REQUIRED)
 ###########
 
 
-## Declare a C++ executable
-add_executable(${PROJECT_NAME}_node
-  src/k4a_ros_bridge_node.cpp
```
Comment (jplapp) [inline CMakeLists.txt:45]: see comment below for reason to remove this

**Review by tonynajjar — APPROVED** [review]
Can you create a ticket to upstream this? Probably will need some changes to upstream

**Comment by jplapp** [comment]
good point, here it is https://lvserv01.logivations.com/browse/AMRNAV-4745

**Pull request merged**